app/lib/pybot_slack_client.rb

```diff
@@ -0,0 +1,32 @@
+class PybotSlackClient
+  include HTTParty
+
+  base_uri ENV['SLACK_SERVICE_URL']
+
+  attr_reader :headers
+
+  def initialize
+    bearer_token = "Bearer #{ENV['SLACK_SERVICE_AUTH_TOKEN']}"
+    @headers = { Authorization: bearer_token }
+  end
+
+  def verify(email)
+    response = self.class.get(
+      '/verify',
+      headers: headers,
+      query: { email: email }
+    )
+
+    response.parsed_response
+  end
+
+  def invite(email)
```

Comment:
This has two states:

No Need to retry
Need to retry

Some of the responses I've seen can be:

No Exception - explicitly allow these get with @AllenAnthes to figure out how he's going to return:
User success invite - 200 from service
User failed invite - user already invited  - 
User failed invite - bad user email thing+mail@gmail.com is an example 

Raise exception For any thing else
User failed invite - error connecting to service - timeout or something
User failed invite - bad service token - 403
User failed invite - service used an invalid token to talk to slack - whatever service sends us

Reply:
After discussion this with @AllenAnthes  raise an exception unless it's a 200 since he's handling all the good or bad things.